File: hash_tables.js
const tap = require("tap");

const voted = {};

const checkVoter = name => {
  if (voted[name]) {
    return "kick them out!";
  } else {
    voted[name] = true;
    return "let them vote!";
  }
};

tap.equal(checkVoter("mike"), "let them vote!");
tap.equal(checkVoter("mike"), "kick them out!");

File: hash_tables.go
package grok

var voted map[string]bool

func init() {
	voted = make(map[string]bool)
}

func checkVoter(name string) string {
	if voted[name] {
		return "kick them out!"
	}
	voted[name] = true
	return "let them vote!"
}
